rfcs/rfc00XX.md

```diff
@@ -0,0 +1,257 @@
+method:
+
+    package My::Module {
+        use v5.38;
+
+        sub import ($class, @args) {
+            ...
+        }
+        sub bar  :export            {...} # @EXPORT_OK
+        sub baz  :export(:strings)  {...} # %EXPORT_TAGS
+        sub bay  :export(:strings)  {...} # %EXPORT_TAGS
+        sub quux :export(:ALWAYS)   {...} # @EXPORT
+        sub whee                    {...} # not exported
+    }
+
+If an `import()` method is defined, it's possible that it might have special
+importing functionality provided, but we will _still_ allowing importing of
+functions with `:export` tags. We pass the import arguments unchanged to the
+`import()` method (in `@args`, above) and the module author will be
+responsible for filtering those manually. This seems unfortunate, but if
+someone uses `use v5XX`, their import mechanism shouldn't suddenly break.
+
+As a counter-argument, it's unlikely someone will be using `:export`
+attributes, so we could simply filter out all subnames so they don't populate
+`@args`, or filter out all subnames tagged with `:export`.
+
+For the above, that means that these are problematic:
+
+    use My::Module ':srtings';
+    use My::Module 'whee';
+
+For `:srtings`, that's almost certainly misspelled. For `whee`, that's not
+supposed to be exported, but perhaps the module author wants to allow that in
+the argument list.
+
+To side-step this, we could create an alternate to the `import()` method,
+perhaps as something analogous to a phaser, but taking arguments, or a new
+`IMPORT` method. Bike-shedding may now commence!
+
+## Security Implications
+
+None anticipated, but by having the imports being lexical, it means outside
+code might have a harder time replacing these functions.
+
+## Examples
+
+    package My::Module {
+        use v5.38;
+
+        sub bar  :export            {...} # @EXPORT_OK
+        sub baz  :export(:strings)  {...} # %EXPORT_TAGS
+        sub bay  :export(:strings)  {...} # %EXPORT_TAGS
+        sub quux :export(:ALWAYS)   {...} # @EXPORT
+        sub whee                    {...} # not exported
+    }
+
+    use My::Package 'bar';     # imports bar() and quux
+
+    use My::Package ':strings' # imports baz, bay, and quux
+
+    use My::Package;           # imports quux
+
+    use My::Package ();        # imports nothing
+
+    use My::Package qw/bar :strings -quux/; # don't import quux
+
+    use My::Module 'whee'; # fatal error
+
+## Prototype Implementation
+
+None that I'm aware of, though
+[Perl6::Export::Attrs](https://metacpan.org/pod/Perl6::Export::Attrs) seems
+the closest.
+
+## Future Scope
+
+We might want to allow variables and constants to be exported. We might always
+want export tags to contain other export tags to allow "grouping" of groups.
+
+The 'is not exported' error message could list what is actually exported by
+the module.
+
+An introspection API for procedural code (similar to a MOP), could be
```

Comment:
Some kind of API would be nice, yes. Combined with the `IMPORT` phaser idea, I suspect a nice way to implement this would be to just make something initially that works, and then create some API functions (perhaps in the `builtin::` space) and state that the default behaviour is equivalent to if you had written some standard example as an IMPORT phaser, using those functions. That way, if people want to understand what the behaviour is, they can read that example code. If they want to implement some slight variation on the theme, they can copy-paste it then edit as appropriate.

In effect, we'd say that the default provided implementation is *as if* the user had written that example block (though, likely internally implemented in a more efficient manner).

Reply:
I like this and will think about it. I would like to see more concrete examples and suggestions, so I won't write anything up yet.

Reply:
Regarding introspection, I kind of like the idea of just reusing the current variables Exporter uses.